packages/langium/src/workspace/documents.ts

```diff
@@ -288,10 +312,28 @@ export interface LangiumDocuments {
     addDocument(document: LangiumDocument): void;
 
     /**
-     * Retrieve the document with the given URI, if present. Otherwise create a new document
-     * and add it to the managed documents.
+     * Retrieve the document with the given URI, if present. Otherwise returns `undefined`.
+     */
+    getDocument(uri: URI): LangiumDocument | undefined;
+
+    /**
+     * Retrieve the document with the given URI. If not present, a new one will be created using the file system access.
+     * The new document will be added to the list of documents managed under this service.
+     */
+    getOrCreateDocument(uri: URI, cancellationToken?: CancellationToken): Promise<LangiumDocument>;
+
+    /**
+     * Creates a new document with the given URI and text content.
+     * The new document is automatically added to this service and can be retrieved using {@link getDocument}.
+     */
```

Comment:
Add a hint that this method throws an error if a document with the given URI is already present (also for the other overload).